app/serializers/concept_serializer.rb

```diff
@@ -0,0 +1,24 @@
+class ConceptSerializer < ActiveModel::Serializer
+  attributes :id, :name, :uid, :parent_id, :level, :parent_uid
+  delegate :cache_key, :to => :object
+
+  def to_json(*args)
+    Rails.cache.fetch expand_cache_key(self.class.to_s.underscore, cache_key, 'to-json') do
+      super
+    end
+  end
+
+  def parent_uid
+    object.parent.try(:uid)
```

Comment:
@kriskelly what do you think about just pulling this in with the SQL? Though I guess it depends on time constraints - this is probably fine for now.